src/realm/object-store/shared_realm.cpp

```diff
@@ -81,7 +81,7 @@ Realm::Realm(Config config, util::Optional<VersionID> version, std::shared_ptr<_
     , m_scheduler(m_config.scheduler)
 {
     if (!coordinator->get_cached_schema(m_schema, m_schema_version, m_schema_transaction_version)) {
-        m_transaction = coordinator->begin_read(version.value_or(VersionID{}));
+        m_transaction = coordinator->begin_read();
```

Comment:
This reverts the erroneous behaviour of using the version used for setting the `frozen_version`, for opening realm